### PR TITLE
Intercept banner

### PR DESCRIPTION
**Pull request by gtk-grafana**
Fixes: https://github.com/grafana/explore-logs/issues/505

image

@stevendungan @zizzpudding @matryer @JStickler Please review my draft copy and let me know if we want to add/remove anything.

We'll need a link to the video as well.

### CODE REVIEW

**Comment by stevendungan** [comment]
@gtk-grafana 
* will the banner re-appear each session or will it remember state after it's been dismissed?
* is it pinned to the top of the page, or will it move with the user as they vertically scroll?

@JStickler @zizzpudding @matryer curious your opinions, i think we could shrink the screen real estate it takes and reword the content a bit but this directionally what we want and close.l

File: src/Components/IndexScene/InterceptBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            Watch this{' '}
+            <a className="external-link" href="#">
+              quick video
+            </a>{' '}
+            to learn how Explore Logs can help investigate your Loki logs without writing a single query!
+          </div>
+          <br />
+
+          <div>
+            We&rsquo;re looking for help in shaping the future of Logs Explore. If you have any issues or suggestions,
```
Comment (JStickler) [inline src/Components/IndexScene/InterceptBanner.tsx:23]: Logs Explore or Explore Logs?

File: src/Components/IndexScene/InterceptBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+              href="https://github.com/grafana/explore-logs?tab=readme-ov-file#explore-logs"
+              className="external-link"
+              target="_blank"
+              rel="noreferrer"
+            >
+              docs
+            </a>{' '}
+            or{' '}
+            <a
+              href="https://github.com/grafana/explore-logs/releases"
```
Comment (JStickler) [inline src/Components/IndexScene/InterceptBanner.tsx:43]: Would it be better to link to a CHANGELOG file?    * looks around *  Doesn't look like this repo has a changelog file (yet?)?

**Comment by JStickler** [comment]
>i think we could shrink the screen real estate it takes and reword the content a bit

I'm OK with the size.  I'll take another look at the text in the morning when I'm fresh and can make good words again.  😛

**Comment by gtk-grafana** [comment]
> * will the banner re-appear each session or will it remember state after it's been dismissed?
> * is it pinned to the top of the page, or will it move with the user as they vertically scroll?
> 
> @JStickler @zizzpudding @matryer curious your opinions, i think we could shrink the screen real estate it takes and reword the content a bit but this directionally what we want and close.l

Its removal is saved in local storage, will show until user closes it. That is, as long as you use the same device, you'll only see it once
Currently pinned to the top of the page, not fixed. I think it's annoying enough as is

**Comment by stevendungan** [comment]
ok, i have no further strong opinions, this does the task well enough. i'm good.

File: src/Components/IndexScene/InterceptBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
+import { Alert } from '@grafana/ui';
+import React from 'react';
+
+export function InterceptBanner(props: { interceptDismissed: boolean; onRemove: () => void }) {
+  return (
+    <>
+      {!props.interceptDismissed && (
+        <Alert
+          severity={'info'}
+          title={'Explore logs is under active development in a preview state.'}
```
Comment (matryer) [inline src/Components/IndexScene/InterceptBanner.tsx:10]: maybe we can say,

"Welcome to Explore Logs (Preview) - we're in active development so please expect things to change.

Check out our video or dive into our Get Started docs for tips on how to use the app.

Let us know what we got right and what we got wrong by sending us your feedback."

Can we put a thumbnail of the video in there (it's fine if it pops up in a new window)?

Comment (matryer) [reply src/Components/IndexScene/InterceptBanner.tsx:10]: there will be a Get Started doc page: https://github.com/grafana/explore-logs/pull/456/files#diff-885722bddf5b70461db4e3b67474d96e8e0e3aafa781f7c3005cc7a38fa91f63

**Comment by zizzpudding** [comment]
A couple of thoughts. 

- Could you add a "don't show me again" ? If someone dismisses it without reading first, will they be able to get back to it?  There are links to lots of good content here, it would be a shame to not have access to it.  Maybe we should also have this information available off of a button in the ui? 
- Does the Github link to a repository of all known issues or does it link to "open a new issue" 
- I like Mat's conversational approach to the message.  I'm not sure with the link to what we got right and what we got wrong.  
- "right" good feedback : slack channel or survey?
- "wrong" bad feedback : GH issue?

**Comment by gtk-grafana** [comment]
@JStickler @matryer Joan and I just met, this is what we came up with:

image
image


Not sure where we'd put the video thumbnail. I'd suggest punting for now.
I'll check with @stevendungan as to if we'll have a video to link to by next week, if not we'll remove that bit.

**Comment by JStickler** [comment]
@gtk-grafana Looks good!

File: src/Components/IndexScene/InterceptBanner.tsx (hunk 1 above)
Comment (matryer) [reply src/Components/IndexScene/InterceptBanner.tsx]: `Explore Logs` for now.

**Comment by matryer** [comment]
@gtk-grafana love how simple that is.

**Comment by matryer** [comment]
@gtk-grafana is this ready for review yet?

**Comment by gtk-grafana** [comment]
@matryer just waiting on the link for "Check out our video" and "Get started doc"

**Comment by JStickler** [comment]
@gtk-grafana  the link to the docs is  `https://grafana.com/docs/grafana-cloud/visualizations/simplified-exploration/logs/   `

**Comment by gtk-grafana** [comment]
Last update: removed video callout, updated links
image

Welcome to Explore Logs (public preview) - we're in active development so please expect things to change.
Check out our [Get started doc](https://grafana.com/docs/grafana-cloud/visualizations/simplified-exploration/logs/), or see [recent changes](https://github.com/grafana/explore-logs/releases).
Help us shape the future of the app. [Send us feedback](https://forms.gle/1sYWCTPvD72T1dPH9) or engage with us on [GitHub](https://github.com/grafana/explore-logs/?tab=readme-ov-file#explore-logs).

File: src/Components/IndexScene/LayoutScene.tsx [hunk 4]
```diff
@@ -1,26 +1,54 @@
...
 
     const styles = useStyles2(getStyles);
     return (
       <div className={styles.bodyContainer}>
+        <InterceptBanner
+          interceptDismissed={interceptDismissed}
+          onRemove={() => {
+            model.dismiss();
+          }}
+        />
```
Comment (svennergr) [inline src/Components/IndexScene/LayoutScene.tsx:51]: I think we rather want to move the `interceptDismissed` logic out of the component, no? 

```suggestion
        {!interceptDismissed && (<InterceptBanner
          onRemove={() => {
            model.dismiss();
          }}
        />)}
```

Comment (gtk-grafana) [reply src/Components/IndexScene/LayoutScene.tsx]: 🫡

File: src/Components/IndexScene/LayoutScene.tsx [hunk 5]
```diff
@@ -1,26 +1,55 @@
...
 import { AppliedPattern, IndexSceneState } from './IndexScene';
 import { css } from '@emotion/css';
 import { GiveFeedbackButton } from './GiveFeedbackButton';
+import { InterceptBanner } from './InterceptBanner';
+
+interface LayoutSceneState extends SceneObjectState {
+  interceptDismissed: boolean;
+}
+
+const interceptBannerStorageKey = 'interceptBannerStorageKey';
```
Comment (svennergr) [inline src/Components/IndexScene/LayoutScene.tsx:15]: Can we prefix this with `pluginJson.id`, like we do with other localstorage keys?
https://github.com/grafana/explore-logs/blob/bfd4682c33ecf987679358102cb222922fc92b0f/src/services/store.ts#L3-L4

File: src/Components/IndexScene/InterceptBanner.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+          <a className={'external-link'} href="https://forms.gle/1sYWCTPvD72T1dPH9">
+            Send us feedback
+          </a>{' '}
+          or engage with us on{' '}
+          <a
+            className={'external-link'}
+            href="https://github.com/grafana/explore-logs/?tab=readme-ov-file#explore-logs"
+          >
+            GitHub
+          </a>
```
Comment (svennergr) [inline src/Components/IndexScene/InterceptBanner.tsx:37]: Adding `_blank` target.

```suggestion
          <a
            className="external-link"
            target="_blank"
            rel="noreferrer noopener"
            href="https://grafana.com/docs/grafana-cloud/visualizations/simplified-exploration/logs/   "
          >
            Get started doc
          </a>
          , or see{' '}
          <a
            target="_blank"
            rel="noreferrer noopener"
            className="external-link"
            href="https://github.com/grafana/explore-logs/releases"
          >
            recent changes
          </a>
          .<br />
          Help us shape the future of the app.{' '}
          <a
            target="_blank"
            rel="noreferrer noopener"
            className="external-link"
            href="https://forms.gle/1sYWCTPvD72T1dPH9"
          >
            Send us feedback
          </a>{' '}
          or engage with us on{' '}
          <a
            className="external-link"
            target="_blank"
            rel="noreferrer noopener"
            href="https://github.com/grafana/explore-logs/?tab=readme-ov-file#explore-logs"
          >
            GitHub
          </a>
```

File: src/Components/IndexScene/LayoutScene.tsx (hunk 5 above)
Comment (gtk-grafana) [reply src/Components/IndexScene/LayoutScene.tsx]: Done

File: src/Components/IndexScene/InterceptBanner.tsx (hunk 6 above)
Comment (gtk-grafana) [reply src/Components/IndexScene/InterceptBanner.tsx]: And done

File: src/Components/IndexScene/InterceptBanner.tsx [hunk 7]
```diff
@@ -24,21 +24,21 @@ export function InterceptBanner(props: { onRemove: () => void }) {
           , or see{' '}
           <a
             className={'external-link'}
```
Comment (svennergr) [inline src/Components/IndexScene/InterceptBanner.tsx:26]: I know this is a nit, but can we change this to just `"eternal-link"` please?

File: src/Components/IndexScene/InterceptBanner.tsx [hunk 8]
```diff
@@ -15,7 +15,7 @@ export function InterceptBanner(props: { onRemove: () => void }) {
           Check out our{' '}
           <a
             className="external-link"
-            target={'_blank'}
+            target="_blank"
             href="https://grafana.com/docs/grafana-cloud/visualizations/simplified-exploration/logs/   "
```
Comment (svennergr) [inline src/Components/IndexScene/InterceptBanner.tsx:19]:
```suggestion
            href="https://grafana.com/docs/grafana-cloud/visualizations/simplified-exploration/logs/"
```

**Review by svennergr — COMMENTED** [review]
Sorry, still some small nits.

File: src/Components/IndexScene/InterceptBanner.tsx (hunk 7 above)
Comment (gtk-grafana) [reply src/Components/IndexScene/InterceptBanner.tsx]: Yup, agreed and fixed

File: src/Components/IndexScene/InterceptBanner.tsx (hunk 8 above)
Comment (gtk-grafana) [reply src/Components/IndexScene/InterceptBanner.tsx]: Fixed

**Review by svennergr — APPROVED** [review]
Raise The Roof Office

**Pull request merged**